dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java

```diff
@@ -166,6 +175,68 @@ public String getUrl(Contentlet contentlet) {
         assertEquals(urlExpected, newContentlet.getMap().get(HTMLPageAssetAPI.URL_FIELD));
         assertFalse(newContentlet.getMap().containsKey(Contentlet.NULL_PROPERTIES));
     }
+    
+    /**
+     * this test creates a widget type that has constant fields and a content of this type.
+     * It updates the constant field values and insures that the new constant field values are both
+     * in the content object and the newly transformed map
+     * @throws Exception
+     */
+    @Test
+    public void Test_Constant_fields_are_in_map() throws Exception {
+        
+        
+        // create a widget and a content of that widget;
+        final ContentType widgetLikeContenType = TestDataUtils.getWidgetLikeContentType();
+        final Contentlet newContentlet = TestDataUtils.getWidgetContent(true, 1, widgetLikeContenType.id());
+        assertNotNull(newContentlet);
+
+        final List<Field> constants = widgetLikeContenType.fields(ConstantField.class);
+        
+        final String constantVar1 = constants.get(0).variable();
+        final String constantVar2 = constants.get(1).variable();
+        
+        // assert our content type has no constant values in it
+        assertTrue(constants.size()>1);
+        assertNull(constants.get(0).values());
+        assertNull(constants.get(1).values());
+        
+        // assert our content Map has no constant values in it
+        Map<String,Object> map = new ContentletToMapTransformer(newContentlet).toMaps().get(0);
+        assertNull(map.get(constantVar1));
+        assertNull(map.get(constantVar2));
+        
+        
+        // update our Content Type constants to have values
+        final List<Field> allFields = new ArrayList<>(widgetLikeContenType.fields());
+        allFields.removeIf(f->constants.contains(f));
+        for(final Field field : constants) {
+            final String value = UUIDGenerator.generateUuid();
+            Field newField = FieldBuilder.builder(field).values(value).build();
+            allFields.add(newField);
+        }
+        ContentType updatedType =APILocator.getContentTypeAPI(APILocator.systemUser()).save(widgetLikeContenType, allFields);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'updatedType' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4843681)